src/operators.jl

```diff
@@ -879,3 +879,88 @@ function merge_vertices!(g::Graph{T}, vs::Vector{U} where {U<:Integer}) where {T
 
     return new_vertex_ids
 end
+
+"""
+	line_graph(g::SimpleGraph)
+Given a graph `g`, return the graph `lg`, whose vertices are integers that enumerate the 
+edges in `g`, and two vertices in `lg` form an edge iff the corresponding edges in `g` 
+share a common endpoint. In other words, edges in `lg` are length-2 paths in `g`. 
+Note that `i ∈ vertices(lg)` corresponds to `collect(edges(g))[i]`. 
+
+# Examples
+```jldoctest
+julia> using Graphs
+
+julia> g = path_graph(5);
+
+julia> lg = line_graph(g)
+{4, 3} undirected simple Int64 graph
+```
+"""
+function line_graph(g::SimpleGraph)::SimpleGraph
```

Comment:
```suggestion
function line_graph(g::SimpleGraph)
```
Julia does not have true return type annotations. If you have a function `f(x)::T` then Julia will just call `convert(T, f(x))`.